digest.go

```diff
@@ -124,6 +124,16 @@ func NewTransportWithHTTPTransport(username, password string, transport *http.Tr
 	return t
 }
 
+// NewTransportWithHTTPRoundTripper creates a new digest transport using the supplied http.RoundTripper interface.
+func NewTransportWithHTTPRoundTripper(username, password string, transport http.RoundTripper) *Transport {
```

Comment:
Is the prefix `New` a common convention?

Reply:
yes there are no constructors in go so this is the convention for constructor like behavior, also there's no method overloading so the `With` is somewhat common as well

Reply:
And I didn't mention this on the code review since `NewTransportWithHTTPTransport` already exists but both `NewTransportWithHTTPTransport` and the new method are not necessary can already create. anew trnasporto doing 
```go
t := &digest.Transport{
		Username:  username,
		Password:  password,
		Transport: transport,
	}
``` 

or even 
```go
t := &Transport{
		username,
		password,
		transport,
	}
```
`New` methods are only needed if there are private fields or added logic but `Transport` has none

Reply:
My personal opinion is that New methods produce slightly better end user API as new fields in the structure would not cause compilation issues. 

Example:
https://github.com/mongodb/atlas-sdk-go/pull/294/commits/0f2d20a23681cedf67ebe852fe18ab869f1d68ae